sys/mutex.c

```diff
@@ -21,8 +22,12 @@ void _mtx_lock(mtx_t *m, const void *waitpt) {
   }
 
   WITH_NO_PREEMPTION {
-    while (m->m_owner != NULL)
-      sleepq_wait(&m->m_owner, waitpt);
+    while (m->m_owner != NULL) {
+      turnstile_t *ts = turnstile_trywait(m);
```

Comment:
@ilikeheaps, you're using `m` as `wchan` here, whereas in current mutex implementation they're using `&m->m_owner`. @cahirwpz, is it OK?